controllers/movieController.js

```diff
@@ -0,0 +1,146 @@
+const models = require('../models')
+
+async function getAllMovies(request, response) {
+    const movies = await models.Movies.findAll({
+        attributes: ['id', 'title', 'releaseDate', 'runTime', 'ratings'],
+        include: [
+            { model: models.Directors, attributes: ['director'] },
+            { model: models.Genres },
+        ],
+    })
+    response.send(movies)
+
+}
+async function movieById(request, response) {
+    var requestInfo = request.params.MovieID
+    const movies = await models.Movies.findAll({
+        where: {
+            id: requestInfo
+        },
+        attributes: ['title', 'releaseDate', 'runTime', 'ratings'],
+        include: [
+            { model: models.Directors },
+            { model: models.Genres },
+        ],
+    })
+    if (movies.length === 0) {
+        response.send('No Movie Found')
+    } else {
+        response.send(movies)
+    }
+}
+
+async function getByDirectorID(request, response) {
+    var requestInfo = request.params.directorId
+    const directorID = await models.Directors.findOne({
+        attributes: ['director'],
+        where: {
+            id: requestInfo
+        },
+        include: [{
+            model: models.Movies,
+            attributes: ['title', 'releaseDate', 'runTime', 'ratings'],
+            include: [{
+                model: models.Genres,
+                attributes: ['genre']
+            }]
+        }],
+    })
+    response.send(directorID)
+}
+async function getByGenre(request, response) {
+    var requestInfo = request.params.genre
+
+    const genre = await models.Genres.findOne({
+        attributes: ['genre'],
+        where: {
+            genre: requestInfo,
+        },
+        include: [{
+            model: models.Movies,
+            attributes: ['title', 'releaseDate', 'runTime', 'ratings'],
+            include: [{
+                model: models.Directors,
+                attributes: ['director']
+            }]
+        }]
+    })
+    response.send(genre)
+}
+async function postMovies(request, response) {
+    const { title, directors, releaseDate, rating, runTime, genres } = request.body
+
+    if (!title || !directors || !releaseDate || !rating || !runTime || !genres) {
+        response.status(400).send('the following are required title, directors, releaseDate, rating, runTime, genres if more then one director or genre seperate with comma')
+    }
+
+    const [movie] = await models.Movies.findOrCreate({
+        where: { title: title, releaseDate: releaseDate, runTime: runTime, ratings: rating },
```

Comment:
I would only put the title in the where. With all of these then it will try to match them all so if someone submitted the same movie with a different run time you'd get two entries. You can use defaults to fill in the rest. Also if the key name and variable name are the same you don't have to provide both.

```JavaScript
  where: { title },
  defaults: { ratings: rating, releaseDate, runTime }
```